### PR TITLE
Backport 15826, 15530, 16243, 15824, 16587, 16555, 16629, 16847, 16857, 14877

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

File: test/functional/feature_block.py [hunk 1]
```diff
@@ -1196,16 +1196,28 @@ def run_test(self):
         self.send_blocks(blocks2, False, request_block=False)
 
         # extend alt chain to trigger re-org
-        block = self.next_block("alt" + str(chain1_tip + 1))
+        block = self.next_block("alt" + str(chain1_tip + 1), version=4)
```
[the remark this answers is not in the document]
Comment (vijaydasmp) [reply test/functional/feature_block.py:1199]: Thanks @UdjinM6 for catching

**Comment by vijaydasmp** [comment]
@PastaPastaPasta @UdjinM6

File: src/rpc/rawtransaction.cpp [hunk 2]
```diff
@@ -1076,7 +1081,7 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
     CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
     const uint256& tx_hash = tx->GetHash();
 
-    CAmount max_raw_tx_fee = ::maxTxFee;
+    CAmount max_raw_tx_fee = DEFAULT_MAX_RAW_TX_FEE;
```
Comment (UdjinM6) [inline src/rpc/rawtransaction.cpp:1084]: 15620: looks incomplete

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -1585,7 +1585,7 @@ static UniValue listsinceblock(const JSONRPCRequest& request)
                 "If \"blockhash\" is no longer a part of the main chain, transactions from the fork point onward are included.\n"
                 "Additionally, if include_removed is set, transactions affecting the wallet which were removed are returned in the \"removed\" array.\n",
                 {
-                    {"blockhash", RPCArg::Type::STR, /* opt */ true, /* default_val */ "", "The block hash to list transactions since"},
+                    {"blockhash", RPCArg::Type::STR, /* opt */ true, /* default_val */ "null", "If set, the block hash to list transactions since"},
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:1588]: 14877: missinig `, otherwise list all transactions.` part

File: src/wallet/rpcwallet.cpp [hunk 4]
```diff
@@ -3349,14 +3349,14 @@ static UniValue fundrawtransaction(const JSONRPCRequest& request)
...
                         {
                             {"changeAddress", RPCArg::Type::STR, /* opt */ true, /* default_val */ "pool address", "The dash address to receive the change"},
                             {"changePosition", RPCArg::Type::NUM, /* opt */ true, /* default_val */ "random", "The index of the change output"},
                             {"includeWatching", RPCArg::Type::BOOL, /* opt */ true, /* default_val */ "false", "Also select inputs which are watch only"},
                             {"lockUnspents", RPCArg::Type::BOOL, /* opt */ true, /* default_val */ "false", "Lock selected unspent outputs"},
                             {"feeRate", RPCArg::Type::AMOUNT, /* opt */ true, /* default_val */ "not set: makes wallet determine the fee", "Set a specific fee rate in " + CURRENCY_UNIT + "/kB"},
-                            {"subtractFeeFromOutputs", RPCArg::Type::ARR, /* opt */ true, /* default_val */ "", "A json array of integers.\n"
+                            {"subtractFeeFromOutputs", RPCArg::Type::ARR, /* opt */ true, /* default_val */ "empty array", "A json array of integers.\n"
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:3359]: 14877: missing https://github.com/bitcoin/bitcoin/pull/14877/files#diff-63c7710d72381cffe692175dd70e47f1891bf75c3e09249ed826f9df3917d02dL3010?

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/wallet/rpcwallet.cpp (hunk 4 above)
Comment (vijaydasmp) [reply src/wallet/rpcwallet.cpp:3363]: Hello @UdjinM6, Thank you, I have made the corrections

**Review by UdjinM6 — DISMISSED** [review]
LGTM, utACK

**Comment by vijaydasmp** [comment]
After merge #4567, my PR had merge conflicts, I have tried to resolve those 
@UdjinM6 @PastaPastaPasta and @dzutte-cpp please confirm if merging looks ok to you

File: src/rpc/rawtransaction.cpp [hunk 5]
```diff
@@ -1484,7 +1484,7 @@ UniValue createpsbt(const JSONRPCRequest& request)
                                 {
                                     {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
                                     {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
-                                    {"sequence", RPCArg::Type::NUM, /* default */ "", "The sequence number"},
+                                    {"sequence", RPCArg::Type::NUM, /* default */ "depends on the value of the 'replaceable' and 'locktime' arguments", "The sequence number"},
```
Comment (UdjinM6) [inline src/rpc/rawtransaction.cpp:1487]: 14877: we have no `replaceable`

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/rpc/rawtransaction.cpp (hunk 5 above)
Comment (UdjinM6) [reply src/rpc/rawtransaction.cpp]: hmm... you dropped it all for some reason but we do have `locktime` and it does affect `sequence`, should rather be `depends on the value of the 'locktime' argument` imo.

**Comment by vijaydasmp** [comment]
My bad , I will correct it

File: src/rpc/rawtransaction.cpp (hunk 5 above)
Comment (vijaydasmp) [reply src/rpc/rawtransaction.cpp]: Hello @UdjinM6 I have corrected it now

**Review by UdjinM6 — APPROVED** [review]
utACK

File: test/functional/test_framework/key.py [hunk 6]
```diff
@@ -1,216 +1,386 @@
...
+        r = R[0] % SECP256K1_ORDER
+        s = (modinv(k, SECP256K1_ORDER) * (z + self.secret * r)) % SECP256K1_ORDER
+        if low_s and s > SECP256K1_ORDER_HALF:
+            s = SECP256K1_ORDER - s
+        # Represent in DER format. The byte representations of r and s have
+        # length rounded up (255 bits becomes 32 bytes and 256 bits becomes 33
+        # bytes).
+        rb = r.to_bytes((r.bit_length() + 8) // 8, 'big')
+        sb = s.to_bytes((s.bit_length() + 8) // 8, 'big')
+        return b'\x30' + bytes([4 + len(rb) + len(sb), 2, len(rb)]) + rb + bytes([2, len(sb)]) + sb
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply test/functional/test_framework/key.py:386]: This file was removed in https://github.com/dashpay/dash/commit/313d79e3190ea19ec572d15284c8077c356be9f7#diff-f3972aca3482ea96a5267ce8f92f6900d18740e6b9f80bd1fd8d115499e1d466 and re-added in https://github.com/dashpay/dash/commit/dbfa92990afc5d8099bcf4f17c585ff8b7997b78#diff-f3972aca3482ea96a5267ce8f92f6900d18740e6b9f80bd1fd8d115499e1d466 but with `--min-confidence 100` and no ignore list (both via #4487), all pre- 313d79e3190ea19ec572d15284c8077c356be9f7 changes can/must be ignored now.

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**